libs/ledger-live-common/src/families/ethereum/synchronisation.ts

```diff
@@ -54,31 +55,55 @@ export const getAccountShape: GetAccountShape = async (
   const blockHashExistsOnChain = await api
     .getBlockByHash(mostRecentStableOperation?.blockHash)
     .then(Boolean);
+
   const syncHash =
     JSON.stringify(blacklistedTokenIds || []) +
     "_" +
     listTokensForCryptoCurrency(currency, {
       withDelisted: true,
     }).length;
+
   const outdatedSyncHash = initialAccount?.syncHash !== syncHash;
+  const allOperationsLoaded = initialAccount
+    ? areAllOperationsLoaded(initialAccount)
+    : false;
+
   const pullFromBlockHeight =
     initialAccount &&
-    areAllOperationsLoaded(initialAccount) &&
+    allOperationsLoaded &&
     mostRecentStableOperation &&
     blockHashExistsOnChain &&
     !outdatedSyncHash
       ? mostRecentStableOperation.blockHeight
       : undefined;
+
   const txsP = fetchAllTransactions(api, address, pullFromBlockHeight);
   const [txs, currentBlock, balance] = await Promise.all([
     txsP,
     currentBlockP,
     balanceP,
   ]);
-  const blockHeight = currentBlock.height.toNumber();
+
+  const blockHeight = currentBlock?.height.toNumber();
+
+  log("Ethereum", "fetchAllTransactions", txs);
+  log("Ethereum", "pullFromBlockHeight conditions", {
+    initialAccount,
+    allOperationsLoaded,
+    mostRecentStableOperation,
+    blockHashExistsOnChain,
+    outdatedSyncHash,
+  });
```

Comment:
`log("Ethereum", "fetchAllTransactions", txs);` can also be problematic, I would suggest only logging the parameters of the function + the txs.length.

```typescript
log("Ethereum", "pullFromBlockHeight conditions", {
    initialAccount,
    allOperationsLoaded,
    mostRecentStableOperation,
    blockHashExistsOnChain,
    outdatedSyncHash,
  });
```
Same here, logging `initialAccount` will be problematic, maybe log only the `initialAccount?.operations?.length` is enough for us. We'll know that the intial account exists, and if it already had a state that shouldn't have been overridden. 

I'm also wondering if we shouldn't only log that in the `no ops` condition 🤔